### PR TITLE
Feature: Tags history

### PR DESCRIPTION
**Pull request by pylipp**
- Add temporal columns to TagsRelation
- Update TagsRelation business logic to set temporal fields when assigning or removing tags
- Update DB seed

### CODE REVIEW

**Comment by pylipp** [comment]
[2.0/dropapp: implement temporal awareness about adding/removing tags](https://trello.com/c/0QOt3EcA/1489-20-dropapp-implement-temporal-awareness-about-adding-removing-tags)

**Review by HaGuesto — APPROVED** [review]
# code reviewed passed

# feature review
We move the feature review to when INserts and Deletions are also covered, because we need an updated init.sql here first.

File: back/boxtribute_server/graph_ql/definitions/protected/types.graphql [hunk 1]
```diff
@@ -46,6 +46,7 @@ type Box implements ItemsCollection {
 }
 
 type HistoryEntry {
+  " IDs are not guaranteed to be unique due to history data being pulled from multiple sources "
```
Comment (HaGuesto) [inline back/boxtribute_server/graph_ql/definitions/protected/types.graphql:49]: please make a consolidation task out of changing this comment: Rather add a prfix to the ID

Comment (pylipp) [reply back/boxtribute_server/graph_ql/definitions/protected/types.graphql]: see [653b0ad](https://github.com/boxwise/boxtribute/pull/1472/commits/653b0ad871ec0fb64765f7983b0b5d85ee51ccd5)

**Comment by HaGuesto** [comment]
# functional review
There is only one case where I'd probably change something. I would not show Box history items about assigning tags on the timestamp when the box is created. We also do not show the selected product at the moment of creation in the box history and this would end up confusing when tags are assigned before the box is created.
image

Therefore, I'd remove the assign tags entries from the box history when the box is created.

**Comment by pylipp** [comment]
I agree that it might be confusing to see the created-record entries as more recent in the history. I thought this might be solved by grouping history entries with identical timestamp (tagged you in [trello](https://trello.com/c/Yur2g6CU)).

If you think this is not sufficient, I'll try to make the assigned-tag entries that belong to creation disappear. I think the easiest way is to not set the created-fields of TagsRelations when creating a box. Then these fields would only be set if an existing box is updated with tags.

**Pull request merged**